src/main/render/shared/includes.ts

```diff
@@ -13,24 +13,38 @@ import { Resolver } from '../../resolver'
 import { DefinitionType, INamespacePath, IRenderState } from '../../types'
 import { COMMON_IDENTIFIERS } from './identifiers'
 
+
+/**
+ * Boolean check for whether a field uses thrift or Int64
+ * @param fieldType Field type we are resolving
+ * @param state Current render state used for resolving definitions
+ * @param recursiveResolve Recursively check field definitions
+ * @param int64Check Flag for Int64 vs thrift import statements. There is most likely a cleaner way of doing
+ * this without duplicating the function body.
+ */
 function fieldTypeUsesThrift(
     fieldType: FieldType,
     state: IRenderState,
+    recursiveResolve: boolean = false,
```

Comment:
I don't know how much we want to stray from the upstream project, but once we get to 3 or 4 params, especially with some that have optional values, it's usually better to use an options with named params instead of a parameter list. Maybe we can do that for at least these two params, and leave fieldType and state as is?

```
function fieldTypeUsesThrift(fieldType: FieldType, state: IRenderState, {recursiveResolve, int64Check}: {recursiveResolve: boolean, int64Check: boolean} = {recursiveResolve: true, int64Check: false}) {
}
```

Reply:
That sounds good to me. At the very least it keeps our new parameters separate. There are a few parts of this that are safe to contribute back to the original project.

There will definitely be drift but I actually think it will be minor after these changes. I don't plan on touching this again until we run into an error. I've thought of a few cases where struct constants have embedded unions may fail but it'll be easy to expand in that case (adding an `||` to some if checks).